pkg/util/watch.go

```diff
@@ -493,6 +428,82 @@ func Watch[C WatchClient[L], L metav1.ListMetaAccessor, T any, P WatchObject[T]]
 	return &store, nil
 }
 
+// helper for Watch. Error events are expected to already have been handled by the caller.
+func handleEvent[T any, P ~*T](
```

Comment:
wasn't this my handcrank change, more or less?

Reply:
yep!

> I think something very similar was originally implemented in https://github.com/neondatabase/autoscaling/pull/65, but that PR seems mostly stalled.